homeassistant/components/lg_thinq/entity.py

```diff
@@ -29,43 +33,42 @@ def __init__(
         self,
         coordinator: DeviceDataUpdateCoordinator,
         entity_description: EntityDescription,
-        property: ThinQProperty,
+        idx: str,
     ) -> None:
         """Initialize an entity."""
         super().__init__(coordinator)
 
         self.entity_description = entity_description
-        self.property = property
+        self.idx = idx
+
         self._attr_device_info = dr.DeviceInfo(
             identifiers={(DOMAIN, coordinator.unique_id)},
             manufacturer=COMPANY,
-            model=coordinator.device_api.model_name,
+            model=coordinator.api.device.model_name,
             name=coordinator.device_name,
         )
-
-        # Set the unique key. If there exist a location, add the prefix location name.
-        unique_key = (
-            f"{entity_description.key}"
-            if property.location is None
-            else f"{property.location}_{entity_description.key}"
-        )
-        self._attr_unique_id = f"{coordinator.unique_id}_{unique_key}"
+        self._attr_unique_id = f"{coordinator.unique_id}_{self.idx}"
+        self._attr_translation_placeholders = {
+            "location": (
+                ""
+                if self.location is None
+                or self.location in (Location.MAIN, Location.OVEN, coordinator.sub_id)
+                else f"{self.location} "
+            )
+        }
```

Comment:
We should only add translation placeholders when they are required. In english having `{location} remote start` makes sense, but in another language that might be really strange. So forcing the placeholder to have a space at the end is a bad practice.

Reply:
Then we don't have any way to show the location?

Reply:
Oh I mean, in those cases we should have a separate translation key
```json
{
    "remote_start": {
        "name": "Remote start"
    },
    "remote_start_for_location": {
        "name": "{location} remote start"
    }
}
```
(also notice that now the casing changes)

Reply:
Modified as guided.